scr/Parking-Zone.Data/IRepositories/IRepository.cs

```diff
@@ -0,0 +1,16 @@
+﻿using Parking_Zone.Domain.Entities;
+
+namespace Parking_Zone.Data.IRepositories;
+
+public interface IRepository<T> where T : Auditable
```

Comment:
I am sorry bro, i cannot change auditable into "class", I understood you if I use class instead of Auditable my repository is going to more generic and general, but auditable is created as base class for my all entites and it contains Id , createdAt and updatedAt properties, if i dont use it, ex : i cannot find entity with id